pkgs/applications/science/biology/psipred/default.nix

```
@@ -0,0 +1,41 @@
+{ lib, stdenv, fetchFromGitHub, blast, tcsh, dbPath ? "uniref90.fasta", }:
+
+stdenv.mkDerivation rec {
+  pname = "psipred";
+  version = "4.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1frxpnrxkhj968k0w6r4npsn5g0jjd0bx6wmb68yqbqkns1qfrgf";
+  };
+
+  buildInputs = [ blast tcsh ];
+
+  prePatch = ''
+    substituteInPlace ./BLAST+/runpsipredplus \
+      --replace "uniref90filt" "${dbPath}" \
+      --replace "/usr/local/bin" "${blast}/bin" \
+      --replace "../bin" "$out/bin" \
+      --replace "../data" "$out/data"
+  '';
+
+  preBuild = ''
+    cd ./src
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin $out/data
+    cp psipred psipass2 chkparse seq2mtx ../BLAST+/runpsipredplus $out/bin
+    cp -r ../data $out/
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out/bin $out/data
    cp psipred psipass2 chkparse seq2mtx ../BLAST+/runpsipredplus $out/bin
    cp -r ../data $out/
    
    runHook postInstall
  '';
```
Also, I suggest using `install -Dm755` instead of mkdir/cp here.

---

Oh, I forgot to add `runHook`.
I did not know about `install`, but I followed your suggestion, thank you!